tools/mpremote/mpremote/main.py

```diff
@@ -293,12 +293,35 @@ def _list_recursive(files, path):
                 if d not in known_dirs:
                     pyb.exec_("try:\n uos.mkdir('%s')\nexcept OSError as e:\n print(e)" % d)
                     known_dirs.add(d)
-            pyboard.filesystem_command(pyb, ["cp", "/".join((dir, file)), ":" + dir + "/"])
+            pyboard.filesystem_command(
+                pyb,
+                ["cp", "/".join((dir, file)), ":" + dir + "/"],
+                progress_callback=show_progress_bar,
+            )
     else:
-        pyboard.filesystem_command(pyb, args)
+        pyboard.filesystem_command(pyb, args, progress_callback=show_progress_bar)
     args.clear()
 
 
+def show_progress_bar(size, total_size):
```

Comment:
I feel like this function should go before `do_filesystem()`, because it's used by it.

Reply:
done